src/aiodynamo/http/base.py

```diff
@@ -0,0 +1,41 @@
+import abc
+from dataclasses import dataclass
+from typing import *
+
+from aiodynamo.types import Timeout
+from yarl import URL
+
+Headers = Mapping[str, str]
+
+
+@dataclass
+class RequestFailed(Exception):
+    url: URL
+    status: int
+    response: Optional[bytes] = None
+    headers: Optional[Headers] = None
+    body: Optional[bytes] = None
+
+
+class HTTP(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    async def get(
+        self, *, url: URL, headers: Optional[Headers] = None, timeout: Timeout
+    ) -> bytes:
+        """
+        Make a GET request and return the response as bytes.
+
+        Raise a RequestFailed exception if the request was not successful.
+        """
+        pass
+
+    @abc.abstractmethod
+    async def post(
+        self, *, url: URL, body: bytes, headers: Optional[Headers] = None
+    ) -> Dict[str, Any]:
+        """
+        Make a POST request and return the parsed JSON object.
+
+        Raise a RequestFailed exception if the request was not successful.
+        """
+        pass
```

Comment:
I'm curious why the API has `get() -> bytes` but `post() -> json` ?

Reply:
Yea that is indeed odd, but I felt for a specialized HTTP client interface it should be fine. The reason is that `post` is used for DynamoDB APIs and the return value is always JSON, so clients that have some form of built-in, possibly optimized, JSON parser can use that (maybe less copying of the response or something). The `get` API is used by credential fetchers and their responses are not always JSON, so that returns `bytes` instead. In the end, using this strange API is a bit nicer, so I think it's worth it.